library/templates/pharmacy_list_form.php

```diff
@@ -0,0 +1,415 @@
+<?php
+
+/**
+ * Handles the creating and updating of the weno alternate and primary pharmacies. 
+ * Saving of these properties are handled by the scripts
+ * that call the LBF form.  The current example can be found in 
+ * interface/patient_file/summary/demographics_save.php
+ * and in interface/new/new_comprehensive_save.php
+ *
+ * @package   OpenEMR
+ * @link      https://www.open-emr.org
+ *
+ * @author    Kofi Appiah <kkappiah@medsov.com>
+ * @copyright Copyright (c) 2023 Omega Systems Group International. <info@omegasystemsgroup.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once $GLOBALS['srcdir'] . '/options.inc.php';
+
+use OpenEMR\Services\PharmacyService;
+
+$widgetConstants = [
+    'listWithAddButton' => 26
+    ,'textDate' => 4
+    ,'textbox' => 2
+];
+
+global $pid; // we need to grab our pid from our global settings.
+$pid = ($frow['blank_form'] ?? null) ? 0 : $pid;
+
+// should always be set, but just in case we will set it to 0 so we can grab it
+$field_id_esc = $field_id_esc ?? '0';
+$name_field_id = "form_" . $field_id_esc;
+$smallform = $smallform ?? '';
+
+$pharmacyService = new PharmacyService;
+$prev_prim_pharmacy = $pharmacyService->getWenoPrimaryPharm($_SESSION['pid']) ?? [];
+$prev_alt_pharmacy = $pharmacyService->getWenoAlternateParm($_SESSION['pid']) ?? [];
+$prev_prim_pharmacy = json_encode($prev_prim_pharmacy);
+$prev_alt_pharmacy = json_encode($prev_alt_pharmacy);
+
+$sql = "SELECT list_id, option_id, title FROM list_options WHERE list_id = 'state'";
+$res = sqlStatement($sql);
+
+?>
+
+<style>
+    .warn {
+        color: red;
+        font-size: 10px;
+    }
+</style>
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+</head>
+<body>
+    <form class="col-12" action="<?php echo $_SERVER['PHP_SELF']; ?>" method="POST">
```

Comment:
should never need to use `$_SERVER['PHP_SELF']` if going to same script, then just put the script name.